core/unix/src/TUnixSystem.cxx

```diff
@@ -1481,20 +1481,24 @@ const char *TUnixSystem::TempDirectory() const
 /// Create a secure temporary file by appending a unique
 /// 6 letter string to base. The file will be created in
 /// a standard (system) directory or in the directory
-/// provided in dir. The full filename is returned in base
+/// provided in dir. Optionally one can provide suffix
+/// append to the final name - like extension ".txt" or ".html".
+/// The full filename is returned in base
 /// and a filepointer is returned for safely writing to the file
 /// (this avoids certain security problems). Returns 0 in case
 /// of error.
 
-FILE *TUnixSystem::TempFileName(TString &base, const char *dir)
+FILE *TUnixSystem::TempFileName(TString &base, const char *dir, const char *suffix)
 {
    char *b = ConcatFileName(dir ? dir : TempDirectory(), base);
```

Comment:
Perhaps factor out the suffix check, since it is used twice.
```suggestion
   const bool hasSuffix = suffix && *suffix;
   char *b = ConcatFileName(dir ? dir : TempDirectory(), base);
```